modules/lvgl/input/lvgl_pointer_input.c

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2023 Fabian Blatz <fabianblatz@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT zephyr_lvgl_pointer_input
+
+#include "lvgl_common_input.h"
+#include <zephyr/drivers/display.h>
+#include <lvgl_display.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(lvgl);
+
+struct lvgl_pointer_input_config {
+	struct lvgl_common_input_config common_config; /* Needs to be first member */
+	bool swap_xy;
+	bool invert_x;
+	bool invert_y;
+};
+
+static void process_input_event(const struct device *dev, struct input_event *evt)
+{
+	const struct lvgl_pointer_input_config *cfg = dev->config;
+	struct lvgl_common_input_data *data = dev->data;
+	lv_disp_t *disp = lv_disp_get_default();
+	struct lvgl_disp_data *disp_data = disp->driver->user_data;
+	struct display_capabilities *cap = &disp_data->cap;
+
+	switch (evt->code) {
+	case INPUT_ABS_X:
+		data->pending_event.point.x = evt->value;
```

Comment:
Use a temporary variable for `data->pending_event.point`, should make everything way more compact.